cli/azd/pkg/project/framework_service_dotnet.go

```diff
@@ -42,16 +42,37 @@ func (dp *dotnetProject) Package(ctx context.Context, progress chan<- string) (s
 }
 
 func (dp *dotnetProject) InstallDependencies(ctx context.Context) error {
-	dotnetCli := tools.NewDotNetCli()
-	if err := dotnetCli.Restore(ctx, dp.config.Path()); err != nil {
+	if err := dp.dotnetCli.Restore(ctx, dp.config.Path()); err != nil {
 		return err
 	}
 	return nil
 }
 
+func (dp *dotnetProject) Initialize(ctx context.Context) error {
+	if err := dp.dotnetCli.InitializeSecret(ctx, dp.config.Path()); err != nil {
+		return err
+	}
+
+	handler := func(ctx context.Context, args ServiceLifecycleEventArgs) error {
+		bicepOutput := args.Args["bicepOutput"].(map[string]tools.AzCliDeploymentOutput)
+		for key, val := range bicepOutput {
+			if err := tools.NewDotNetCli().SetSecret(ctx, key, fmt.Sprint(val.Value), dp.config.Path()); err != nil {
+				return err
+			}
```

Comment:
Isn't hte `dotnet` cli already available as an instance member on the struct? You shouldn't need to create a new one here.